### DISCUSSION TITLE
How does RegexSet cache programs?

### DISCUSSION THREAD

**Discussion by adamchalmers** [rust-lang/regex#874] [Q&A] [answered]
Hi! I'm trying to find the source of some memory leaks in a gRPC server my team is building. One of the endpoints runs some RegexSets over some inputs. 

I'm using [dhat](docs.rs/dhat) to measure memory usage, and I notice running `RegexSet::matches(field).into_iter().next()` adds around 2.4kb of memory leak per run. I know the RegexSets have an internal cache where they store programs. I was wondering if this cache grows unboundedly? Is it possible to put a limit on the cache size, or to describe the cache behaviour in the crate's docs?

**Comment by BurntSushi** [comment, marked as answer]
I think it would be better to treat this as a bug report. So I've created an issue from this discussion and responded there: https://github.com/rust-lang/regex/issues/875
